examples/with-reasonml/components/Context.re

```diff
@@ -0,0 +1,27 @@
+module type Config = {
```

Comment:
What does this file mean?  It was quite difficult to explain this to a non-react user.

Reply:
This was the example I found in the discord channel. As far as what it means, it would probably require some discussion on functors in Ocaml/ReasonML but I would argue this file is not as important as `CountContext.re` which provides the API for how to create a global value using the context API. I could always add comments to this file that explains things more.

Reply:
Yes I agree that it will take some discussion of Functors, and module composition (because of the `include`).  Also, this requires a lot of knowledge of React internals.   

I think this is rather scary for someone new.  I like this example because it does showcase some more advanced functionality.  But perhaps it should be it’s own example?  Removing the `ref` from `with-reasonml` and making the context example being `with-reasonml-context` ?

Reply:
Yeah, this is fine having it in a separate example. I am not that attached like I said before. It's was definitely interesting exploring the context API and how to implement it here.